sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/RegexpExpressionsSuite.scala

```diff
@@ -90,6 +105,28 @@ class RegexpExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation("a\nb" like regEx, true, create_row("a%b"))
 
     checkEvaluation(Literal.create(null, StringType) like regEx, null, create_row("bc%"))
+
+    checkEvaluation("" like regEx, true, create_row(""))
+    checkEvaluation("a" like regEx, false, create_row(""))
+    checkEvaluation("" like regEx, false, create_row("a"))
+
+    checkEvaluation("""\\\\""" like regEx, true, create_row("""%\\%"""))
+    checkEvaluation("""%%""" like regEx, true, create_row("""%%"""))
+    checkEvaluation("""\__""" like regEx, true, create_row("""\\\__"""))
+    checkEvaluation("""\\\__""" like regEx, false, create_row("""%\\%\%"""))
+    checkEvaluation("""_\\\%""" like regEx, false, create_row("""%\\"""))
+
+    checkEvaluation("a\u20ACa" like regEx, true, create_row("_\u20AC_"))
+    checkEvaluation("a€a" like regEx, true, create_row("_€_"))
+    checkEvaluation("a€a" like regEx, true, create_row("_\u20AC_"))
+    checkEvaluation("a\u20ACa" like regEx, true, create_row("_€_"))
+
+    checkEvaluation("""a\""" like regEx, false, create_row("""a\"""))  // TODO: should throw an exception?
+
+    checkEvaluation("A" like regEx, false, create_row("a%"))
+    checkEvaluation("a" like regEx, false, create_row("A%"))
+    checkEvaluation("AaA" like regEx, true, create_row("_a_"))
+
```

Comment:
yeah, I just saw those.  I turns out that `scalastyle` is a scoped task, so you have to run `test:scalastyle` in sbt to make sure everything gets checked.